CoreFoundation/Base.subproj/CFPlatform.c

```diff
@@ -558,15 +559,14 @@ CF_EXPORT void _NS_pthread_setname_np(const char *name) {
     }
 }
 
-static pthread_t __initialPthread = { NULL, 0 };
+static _CFThreadRef __initialPthread = INVALID_HANDLE_VALUE;
 
 CF_EXPORT int _NS_pthread_main_np() {
-    pthread_t me = pthread_self();
-    if (NULL == __initialPthread.p) {
-        __initialPthread.p = me.p;
-        __initialPthread.x = me.x;
-    }
-    return (pthread_equal(__initialPthread, me));
+    if (__initialPthread == INVALID_HANDLE_VALUE)
+      DuplicateHandle(GetCurrentProcess(), GetCurrentThread(),
```

Comment:
What are the threading characteristics of this call? The review is conditional on this being thread-safe, or at least harmless if done more than once.

Reply:
The handle is process-wide, so, it is safe.  Calling `GetCurrentProcess` and `GetCurrentThread` is perfectly fine, the former returns the process handle, the latter the current thread's handle.  There is a note above the function that it must be called on the main thread only before any threads are created.

Reply:
Spoke with @millenomi offline, the concern was over this being called from multiple threads, which is not permitted.  This is invoked from `__CFInitialize` before any threads are created, and thus guaranteed to be executed on the main thread.